src/validate.js

```diff
@@ -3100,89 +3103,171 @@ function F_VALIDATE(disabledHL) {
+				// add selected venue to the array
+				selectedObjects.push(venueID);
+
+				// mark venue to revalidate
+				_RT.$revalidate[venueID] = true;
+				// recheck selected venue
+				if (seen) {
+					deleteSeenObject(venueID);
+					seen = null;
+				}
 			}
-		}
-		else {
-			// recheck the venue to revalidate
-			if (segmentID in _RT.$revalidate) {
-				deleteSeenObject(venueID);
-				seen = null;
-				// unmark venue
-				delete _RT.$revalidate[segmentID];
+			else {
+				// recheck the venue to revalidate
+				if (segmentID in _RT.$revalidate) {
+					deleteSeenObject(venueID);
+					seen = null;
+					// unmark venue
+					delete _RT.$revalidate[segmentID];
+				}
 			}
-		}
-
-		// check if the venue was already seen
-		if (seen) {
-			HLObject(rawVenue);
-			continue;
-		}
-
-		///////////////////////////////////////////////////////////////////
-		// Prepare simple objects
-		var venue = new SimpleOBJECT(venueID, WMo.venues);
-		Object.seal(venue);
-
-		// mark venue as seen
-		_RT.$seen[venueID] = seen = [0, null, false, false,
-			4 > currentZoom,
-			cityID];
-
-		// increase city counter
-		venue.incCityCounter();
-
-		// shortcuts
-		var address = venue.$address;
-		var country = address.$country;
-		var countryLen = country.length;
-		var countryCode = country ? _I18n.getCountryCode(country.toUpperCase())
-			: _RT.$cachedTopCCode;
-		var city = address.$city;
-		var cityLen = city.length;
-		var cityID = address.$cityID;
-		var street = address.$street;
-		var state = address.$state;
-		var streetLen = street.length;
-		var alts = venue.$alts;
 
-		var exceptedCategories ='BRIDGE|ISLAND|FOREST_GROVE|SEA_LAKE_POOL|RIVER_STREAM|CANAL|DAM|TUNNEL|JUNCTION_INTERCHANGE'.split('|');
-
-		if (!cityLen
-			&& exceptedCategories.indexOf(venue.$categories[0]) === -1
-			&& isLimitOk(250)
-			&& address.isOkFor(250))
-			venue.report(250);
+			// check if the venue was already seen
+			if (seen) {
+				HLObject(rawVenue);
+				continue;
+			}
 
-	} // for all venues
+			///////////////////////////////////////////////////////////////////
+			// Prepare simple objects
+			var venue = new SimpleOBJECT(venueID, WMo.venues);
+			Object.seal(venue);
+
+			// mark venue as seen
+			_RT.$seen[venueID] = seen = [0, null, false, false,
+				4 > currentZoom,
+				cityID];
+
+			// increase city counter
+			venue.incCityCounter();
+
+			// shortcuts
+			var address = venue.$address;
+			var country = address.$country;
+			var countryLen = country.length;
+			var countryCode = country ? _I18n.getCountryCode(country.toUpperCase())
+				: _RT.$cachedTopCCode;
+			var city = address.$city;
+			var cityLen = city.length;
+			var cityID = address.$cityID;
+			var street = address.$street;
+			var state = address.$state;
+			var streetLen = street.length;
+			var alts = venue.$alts;
+			var lock = venue.$lock;
+
+			var exceptedCategories ='BRIDGE|ISLAND|FOREST_GROVE|SEA_LAKE_POOL|RIVER_STREAM|CANAL|DAM|TUNNEL|JUNCTION_INTERCHANGE'.split('|');
```

Comment:
No hard-coded values should be in the code. Those categories should be moved either to `i18n` as a default value or to the `data.js`.